gap/attributes/attr.gi

```diff
@@ -777,10 +777,42 @@ InstallMethod(InversesOfSemigroupElementNC,
 "for a semigroup and a multiplicative element",
 [IsSemigroup, IsMultiplicativeElement],
 function(S, x)
-  if not IsFinite(S) then
-    TryNextMethod();
-  fi;
-  return Filtered(EnumeratorSorted(S), y -> x * y * x = x and y * x * y = y);
+    if not IsFinite(S) then
+        TryNextMethod();
+    fi;
+    return Filtered(EnumeratorSorted(S), 
+    y -> x * y * x = x and y * x * y = y);
+end);
+
+InstallMethod(InversesOfSemigroupElementNC,
+"for CanUseFroidurePin and a multiplicative element",
+[CanUseFroidurePin, IsMultiplicativeElement],
+function(S, a)
+  local R, L, inverses, e, f, s;
+  R := RClass(S, a);
+  L := LClass(S, a);
+  inverses := [];
+
+  for e in Idempotents(R) do
+    s := RightGreensMultiplierNC(S, a, e) * e;
+    for f in Idempotents(L) do
+      Add(inverses, f * s);
+    od;
+  od;
+  return inverses;
+end);
+
+InstallMethod(OneInverseOfSemigroupElementNC,
```

Comment:
It'd be good to have a version `OneInverseOfSemigroupElement` i.e. without the `NC`, and documentation for that.